onnxruntime/core/providers/xnnpack/math/matmul.cc

```diff
@@ -41,9 +41,9 @@ bool MatMul::IsOnnxNodeSupported(const NodeUnit& node_unit, const GraphViewer& g
       break;
     }
 
-    if (A_shape == nullptr || A_shape->dim_size() > 2 ||
-        (A_shape->dim_size() == 2 && A_shape->dim(1).dim_value() == 0) ||
```

Comment:
All the gating logic around a dim alue being 0 should be taken care of in Compute() because we bail out if the output buffer's size is going to be 0